From 5f1b41d840b8893c0a0d88295433554dfc3b2469 Mon Sep 17 00:00:00 2001
From: Lily Zhang <r58066@freescale.com>
Date: Thu, 8 Apr 2010 18:00:05 +0800
Subject: [PATCH] ENGR00122359 MX53: Change FEC PAD setting

FEC can not work well if plugging into auto port card.
Change FEC PAD setting to fix it.

Signed-off-by: Lily Zhang <r58066@freescale.com>
---
 arch/arm/mach-mx5/mx53_evk_gpio.c |   13 +++++++------
 1 files changed, 7 insertions(+), 6 deletions(-)

diff --git a/arch/arm/mach-mx5/mx53_evk_gpio.c b/arch/arm/mach-mx5/mx53_evk_gpio.c
index 7ca8e77..37c489a 100644
--- a/arch/arm/mach-mx5/mx53_evk_gpio.c
+++ b/arch/arm/mach-mx5/mx53_evk_gpio.c
@@ -463,29 +463,30 @@ static struct mxc_iomux_pin_cfg __initdata mxc_iomux_pins[] = {
 	 },
 	{ /* FEC pins */
 	 MX53_PIN_FEC_MDIO, IOMUX_CONFIG_ALT0,
-	 PAD_CTL_SRE_FAST,
+	 (PAD_CTL_HYS_ENABLE | PAD_CTL_PKE_ENABLE | PAD_CTL_PUE_PULL |
+	  PAD_CTL_22K_PU | PAD_CTL_ODE_OPENDRAIN_ENABLE | PAD_CTL_DRV_HIGH),
 	 MUX_IN_FEC_FEC_MDI_SELECT_INPUT,
 	 INPUT_CTL_PATH1,
 	 },
 	{
 	 MX53_PIN_FEC_REF_CLK, IOMUX_CONFIG_ALT0,
-	 (PAD_CTL_PKE_ENABLE | PAD_CTL_PUE_PULL),
+	 (PAD_CTL_HYS_ENABLE | PAD_CTL_PKE_ENABLE),
 	 },
 	{
 	 MX53_PIN_FEC_RX_ER, IOMUX_CONFIG_ALT0,
-	 (PAD_CTL_PKE_ENABLE | PAD_CTL_PUE_PULL),
+	 (PAD_CTL_HYS_ENABLE | PAD_CTL_PKE_ENABLE),
 	 },
 	{
 	 MX53_PIN_FEC_CRS_DV, IOMUX_CONFIG_ALT0,
-	 (PAD_CTL_PKE_ENABLE | PAD_CTL_PUE_PULL),
+	 (PAD_CTL_HYS_ENABLE | PAD_CTL_PKE_ENABLE),
 	 },
 	{
 	 MX53_PIN_FEC_RXD1, IOMUX_CONFIG_ALT0,
-	 (PAD_CTL_PKE_ENABLE | PAD_CTL_PUE_PULL),
+	 (PAD_CTL_HYS_ENABLE | PAD_CTL_PKE_ENABLE),
 	 },
 	{
 	 MX53_PIN_FEC_RXD0, IOMUX_CONFIG_ALT0,
-	 (PAD_CTL_PKE_ENABLE | PAD_CTL_PUE_PULL),
+	(PAD_CTL_HYS_ENABLE | PAD_CTL_PKE_ENABLE),
 	 },
 	{
 	 MX53_PIN_FEC_TX_EN, IOMUX_CONFIG_ALT0,
-- 
1.5.4.4

